src/libraries/CauldronLib.sol

```diff
@@ -79,14 +84,83 @@ library CauldronLib {
         // 18 + 8 + 5 - 5 - 8 - 10 = 8 decimals
         IERC20 collateral = cauldron.collateral();
         uint256 collateralPrecision = 10 ** collateral.safeDecimals();
-        liquidationPrice = (borrowValue * collateralPrecision ** 2 * 1e5) / COLLATERALIZATION_RATE / collateralAmount / PRECISION;
-        healthFactor = PRECISION - (PRECISION * liquidationPrice * getOracleExchangeRate(cauldron)) / collateralPrecision ** 2;
+        liquidationPrice =
+            (borrowValue * collateralPrecision ** 2 * 1e5) /
+            COLLATERALIZATION_RATE /
+            collateralAmount /
+            EXCHANGE_RATE_PRECISION;
+        healthFactor =
+            EXCHANGE_RATE_PRECISION -
+            (EXCHANGE_RATE_PRECISION * liquidationPrice * getOracleExchangeRate(cauldron)) /
+            collateralPrecision ** 2;
+    }
+
+    /// @notice the liquidator will get "MIM borrowPart" worth of collateral + liquidation fee incentive but borrowPart needs to be adjusted to take in account
+    /// the sSpell distribution taken off the liquidation fee.
+    /// @return collateralAmount Amount of collateral that the liquidator will receive
+    /// @return borrowAmount MIM amount that the liquidator will need to pay back to get the collateralShare
+    function getLiquidationCollateralAndBorrowAmount(
+        ICauldronV2 cauldron,
+        uint256 borrowPart
+    ) internal view returns (uint256 collateralAmount, uint256 borrowAmount) {
+        uint256 exchangeRate = getOracleExchangeRate(cauldron);
+        Rebase memory totalBorrow = getTotalBorrowWithAccruedInterests(cauldron);
+        IBentoBoxV1 box = IBentoBoxV1(cauldron.bentoBox());
+        Rebase memory bentoBoxTotals = box.totals(cauldron.collateral());
+
+        // convert borrowPart to debt
+        borrowAmount = totalBorrow.toElastic(borrowPart, false);
+
+        // how much collateral share the liquidator will get from the given borrow amount
+        collateralAmount = box.toAmount(
+            cauldron.collateral(),
+            // collateralShare
+            bentoBoxTotals.toBase(
+                (borrowAmount * cauldron.LIQUIDATION_MULTIPLIER() * exchangeRate) /
+                    (LIQUIDATION_MULTIPLIER_PRECISION * EXCHANGE_RATE_PRECISION),
+                false
+            ),
+            false
+        );
+
+        // add the sSpell distribution part
+        borrowAmount +=
+            ((((borrowAmount * cauldron.LIQUIDATION_MULTIPLIER()) / LIQUIDATION_MULTIPLIER_PRECISION) - borrowAmount) * DISTRIBUTION_PART) /
```

Comment:
Is there ever a world where we want DISTRIBUTION_PART to be dynamic instead of constant?

Reply:
yes it could, but I'll just redeploy a new contract in this case